generators/upgrade/index.js

```diff
@@ -295,24 +306,40 @@ module.exports = UpgradeGenerator.extend({
         mergeChangesBack() {
             this.log(`Merging changes back to ${this.sourceBranch}...`);
             const done = this.async();
-            this.gitExec(['merge', '-q', UPGRADE_BRANCH], (code, msg, err) => {
-                this.log(chalk.green('Merge done!'));
+            this.gitExec(['merge', '-q', UPGRADE_BRANCH], { silent: this.silent }, (code, msg, err) => {
+                this.success('Merge done!');
                 done();
             });
         }
     },
 
+    checkConflictsInPackageJson() {
+        const done = this.async();
+        this.gitExec(['diff', '--name-only', '--diff-filter=U', 'package.json'], { silent: this.silent }, (code, msg, err) => {
+            if (code !== 0) this.error(`Unable to check for conflicts in package.json:\n${msg} ${err}`);
+            if (msg) {
+                const installCommand = this.clientPackageManager === 'yarn' ? 'yarn' : 'npm install';
+                this.warning(`There are conflicts in package.json, please fix them and then run ${installCommand}`);
+                this.skipInstall = true;
+            }
+            done();
+        });
+    },
+
     install() {
         const done = this.async();
         if (!this.skipInstall) {
-            shelljs.rm('-rf', 'node_modules');
             this.log('Installing dependencies, please wait...');
+            this.log.info('Removing the node_modules directory');
```

Comment:
BTW, here I don't think it really makes sense to delete `node_modules`, yarn is supposed to manage deps better and probably npm5 too. IMO, it just slows down the upgrade process.